### PR TITLE
[Office365GroupsMail] Open source connector

### PR DESCRIPTION
**Pull request by lnborbolla42**
Adding the following operations to the open source repo:

- `ListConversations`
- `GetGroupConversation`
- `ListConversationThreads`
- `ListGroupThreads`
- `GetConversationThread`
- `DeleteConversationThread`
- `ListThreadPosts`
- `GetThreadPost`
- `GetAttachments`
- `ListGroups`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

File: certified-connectors/Office365GroupsMail/apiProperties.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "triggers"
+    ],
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "aad",
+          "clientId": "<<Client ID placeholder>>",
+          "scopes": [
+            "Group.ReadWrite.All offline_access"
```
Comment (shyronosov-microsoft) [inline certified-connectors/Office365GroupsMail/apiProperties.json:14]: Isn't Group.Read.All enough?

File: certified-connectors/Office365GroupsMail/apiProperties.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            "tenantId": {
+              "value": "common"
+            },
+            "resourceUri": {
+              "value": "https://graph.microsoft.com/"
+            }
+          }
+        }
+      },
+      "token:TenantId": {
```
Comment (shyronosov-microsoft) [inline certified-connectors/Office365GroupsMail/apiProperties.json:34]: Is token:TenantId required to be here?

File: certified-connectors/Office365GroupsMail/readme.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+- `GetConversationThread`: Get a specific thread that belongs to a group.
+- `DeleteConversationThread`: Deletes the specified conversation thread.
+- `ListThreadPosts`: Get all the posts of the specified conversation thread.
+- `GetThreadPost`: Retrieves a post in a specified thread.
+- `GetAttachments`: Retrieve a list of attachments that belong to a post.
+- `ListGroups`: Retrieves all Office 365 Groups the user is part of (operation with `internal` visibility used to populate the dropdowns).
+
+## Notes
+
+If you want that all users are able to sign in (and use the connector), instead of only admins, you can gran admin consent to the application. Follow instructions on how to grant admin consent to an application [[here](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/grant-admin-consent)].
```
Comment (shyronosov-microsoft) [inline certified-connectors/Office365GroupsMail/readme.md:56]: typo in `grant`

File: certified-connectors/Office365GroupsMail/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "@odata.nextLink"
+                }
+            }
+        },
+        "/v1.0/me/memberOf/$/microsoft.graph.group": {
+            "get": {
+                "operationId": "ListGroups",
+                "description": "Retrieves all Office 365 Groups the user is part of.",
```
Comment (shyronosov-microsoft) [inline certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:363]: Should it be Microsoft instead of Office?

Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:363]: True! I'll update all the occurrences.

File: certified-connectors/Office365GroupsMail/apiProperties.json (hunk 1 above)
Comment (shyronosov-microsoft) [reply certified-connectors/Office365GroupsMail/apiProperties.json:14]: Maybe for this operation? DeleteConversationThread

File: certified-connectors/Office365GroupsMail/readme.md (hunk 3 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/readme.md:56]: Fixed, thanks!

File: certified-connectors/Office365GroupsMail/apiProperties.json (hunk 1 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiProperties.json:14]: Yes, I'll remove the `offline_access` scope in the next iteration.

Comment (shyronosov-microsoft) [reply certified-connectors/Office365GroupsMail/apiProperties.json:14]: So ReadWrite is needed for DeleteConversationThread?

File: certified-connectors/Office365GroupsMail/apiProperties.json (hunk 2 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiProperties.json:34]: Looks like it is not necessary; I'll remove it.

File: certified-connectors/Office365GroupsMail/apiProperties.json (hunk 1 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiProperties.json:14]: The operation `DeleteConversationThread` needs `Group.ReadWrite.All`; [here](https://docs.microsoft.com/en-us/graph/api/group-delete-thread?view=graph-rest-1.0&tabs=http#permissions) is the documentation.

Apologies, I did not refresh the page and I skipped your last comment.

File: certified-connectors/Office365GroupsMail/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            },
+            "resourceUri": {
+              "value": "https://graph.microsoft.com/"
+            }
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#EB3C00",
+    "publisher": "Microsoft Corporation"
```
Comment (ShyamSunder90) [inline certified-connectors/Office365GroupsMail/apiProperties.json:36]: For consistenacy, lets use Microsoft

File: certified-connectors/Office365GroupsMail/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,768 @@
...
+                        }
+                    },
+                    "404": {
+                        "description": "Invalid group",
+                        "schema": {
+                            "type": "object"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
```
Comment (ShyamSunder90) [inline certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:54]: How are you going to make paging work without policy or a policy template?

File: certified-connectors/Office365GroupsMail/readme.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+- `ListConversations`: Get all the conversations in this group.
+- `GetGroupConversation`: Retrieves the properties of a particular conversation.
+- `ListConversationThreads`: Get all the threads in a group conversation.
+- `ListGroupThreads`: Get all the threads of a group.
+- `GetConversationThread`: Get a specific thread that belongs to a group.
+- `DeleteConversationThread`: Deletes the specified conversation thread.
+- `ListThreadPosts`: Get all the posts of the specified conversation thread.
+- `GetThreadPost`: Retrieves a post in a specified thread.
+- `GetAttachments`: Retrieve a list of attachments that belong to a post.
+- `ListGroups`: Retrieves all Office 365 Groups the user is part of (operation with `internal` visibility used to populate the dropdowns).
```
Comment (ShyamSunder90) [inline certified-connectors/Office365GroupsMail/readme.md:52]: Microsoft365?

File: certified-connectors/Office365GroupsMail/apiDefinition.swagger.json (hunk 6 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:54]: My understanding is that `x-ms-pageable` does not needs neither of those. AzureAD uses it [[here](https://github.com/microsoft/PowerPlatformConnectors/blob/47eb4d2b5552594233b9e9feab366ff494a14ad7/certified-connectors/AzureAD/apiDefinition.swagger.json#L313)] and Office 365 Groups [[here](https://github.com/microsoft/PowerPlatformConnectors/blob/47eb4d2b5552594233b9e9feab366ff494a14ad7/custom-connectors/Office365Groups/apiDefinition.swagger.json#L327)]; am I missing something?

It is also defined in the apiDefinition.swagger schema [[here](https://github.com/microsoft/PowerPlatformConnectors/blob/47eb4d2b5552594233b9e9feab366ff494a14ad7/schemas/apiDefinition.swagger.schema.json#L346)].

File: certified-connectors/Office365GroupsMail/apiProperties.json (hunk 5 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiProperties.json:36]: Got it, I saw others using `Microsoft Corporation` (e.g. AzureIoTCentral and Partner Center Referrals), I'll change it in the next iteration :)

File: certified-connectors/Office365GroupsMail/readme.md (hunk 7 above)
Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/readme.md:52]: True, I'll update this in the next iteration 👍🏼

File: certified-connectors/Office365GroupsMail/apiDefinition.swagger.json (hunk 6 above)
Comment (shyronosov-microsoft) [reply certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:54]: AzureAD uses policy template updatenextlink:
https://github.com/microsoft/PowerPlatformConnectors/blob/47eb4d2b5552594233b9e9feab366ff494a14ad7/certified-connectors/AzureAD/apiProperties.json

Comment (shyronosov-microsoft) [reply certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:54]: If we have the next link property name, we can just add the same in this case.

Comment (lnborbolla42) [reply certified-connectors/Office365GroupsMail/apiDefinition.swagger.json:60]: True, I have added the policy and tested the pagination. Thanks for catching this!

**Pull request merged**